app/services/find/course_type_answer_determiner.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Find
+  class CourseTypeAnswerDeterminer
+    attr_reader :age_group, :visa_status, :university_degree_status
+
+    def initialize(age_group:, visa_status:, university_degree_status:)
+      @age_group = age_group
+      @visa_status = visa_status
+      @university_degree_status = university_degree_status
+    end
+
+    def show_undergraduate_courses?
+      no_further_education? && university_degree_status.present? && no_degrees? && does_not_require_visa_sponsorship?
+    end
+
+    def non_eligible_for_undergraduate_courses?
```

Comment:
```suggestion
    def not_eligible_for_undergraduate_courses?
```

Reply:
Actually, wondering why you chose to name this `non_eligible_for_undergraduate_courses?` instead of `eligible_for_undergraduate_courses?`

If we want to use the inverse of this method it would be confusing, e.g: `!non_eligible_for_undergraduate_courses?` instead of `eligible_for_undergraduate_courses?`

Reply:
Yes, I gave a lot of thought and initially I went with your suggestion, but `eligible_for_undegraduate_courses?` does make sense when there is postgraduate in the middle.

It is because it is the specific and rare deterministic scenario, the way I added this check, will happen when:

1. User is at the current cycle and there is no undergraduate question
2. User is at the next cycle and there is the undergraduate question
3. The user might have a degree, so make `eligible_for_undegraduate_courses?`  to return true will make things confusing
4. The user does not have a degree and needs visa for undergraduate courses then we show the exit page. This is so specific that I chosen to be like that.

So on visa status page I need to redirect to the "You are not eligible" depending on the prior answers.

Reply:
If you feel strongly, I could rename the `non_elibible_for_undergraduate_courses?` to just 

`show_not_eligible_for_undergraduate_exit_page?`  or something similar?

Let me know what you think, for now I will rename to `not_elibible_for_undergraduate_courses?`

Reply:
Done! I follow Lori's suggestion on renaming to show_exit_page?